Mathlib/RingTheory/LocalRing/Subring.lean

```diff
@@ -0,0 +1,286 @@
+/-
+Copyright (c) 2024 Andrew Yang, Yaël Dillies, Javier López-Contreras. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Yaël Dillies, Javier López-Contreras
+-/
+import Mathlib.RingTheory.Ideal.Over
+import Mathlib.RingTheory.Valuation.ValuationSubring
```

Comment:
This import feels like it shouldn't appear in a file with this name.
Can you please split the current file into `LocalRing/Subring` and `LocalRing/Valuation`?

Reply:
I moved the `isMax` results to `Valuation/LocalSubring` (ideally it should live in `Valuation/ValuationSubring` but that file is already quite long)